pkg/libovsdb/encoding_test.go

```diff
@@ -2,48 +2,214 @@ package libovsdb
 
 import (
 	"encoding/json"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
-// empty Set test
-func TestEmptySet(t *testing.T) {
-	emptySet, err := NewOvsSet([]string{})
-	assert.Nil(t, err)
-	jsonStr, err := json.Marshal(emptySet)
-	assert.Nil(t, err)
-	expected := "[\"set\",[]]"
-	assert.JSONEqf(t, expected, string(jsonStr), "they should be equal\n")
-}
-
-// test Set
-func TestSet(t *testing.T) {
-	ovsSet, err := NewOvsSet([]string{"aa", "bb"})
-	assert.Nil(t, err)
-	jsonStr, err := json.Marshal(ovsSet)
-	assert.Nil(t, err)
-	expected := "[\"set\",[\"aa\",\"bb\"]]"
-	require.JSONEqf(t, expected, string(jsonStr), "they should be equal\n")
-}
-
-// empty Map test
-func TestEmptyMap(t *testing.T) {
-	emptyMap, err := NewOvsMap(map[string]string{})
-	assert.Nil(t, err)
-	jsonStr, err := json.Marshal(emptyMap)
-	assert.Nil(t, err)
-	expected := "[\"map\",[]]"
-	assert.JSONEqf(t, expected, string(jsonStr), "they should be equal\n")
-}
-
-// test Map
-func TestMap(t *testing.T) {
-	ovsMap, err := NewOvsMap(map[string]string{"one": "first", "two": "second"})
-	assert.Nil(t, err)
-	jsonStr, err := json.Marshal(ovsMap)
-	assert.Nil(t, err)
-	expected := "[\"map\",[[\"one\",\"first\"],[\"two\",\"second\"]]]"
-	assert.JSONEqf(t, expected, string(jsonStr), "they should be equal\n")
+type marshalSetTestTuple struct {
+	objInput           interface{}
+	jsonExpectedOutput string
+}
+
+type marshalMapsTestTuple struct {
+	objInput           map[string]string
+	jsonExpectedOutput string
+}
+
+var validUUIDStr0 = `00000000-0000-0000-0000-000000000000`
+var validUUIDStr1 = `11111111-1111-1111-1111-111111111111`
+var validUUID0 = UUID{GoUUID: validUUIDStr0}
+var validUUID1 = UUID{GoUUID: validUUIDStr1}
+
+var setTestList = []marshalSetTestTuple{
+	{
+		objInput:           []string{},
+		jsonExpectedOutput: `["set",[]]`,
+	},
+	{
+		objInput:           `aa`,
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           false,
+		jsonExpectedOutput: `false`,
+	},
+	{
+		objInput:           10,
+		jsonExpectedOutput: `10`,
+	},
+	{
+		objInput:           10.2,
+		jsonExpectedOutput: `10.2`,
+	},
+	{
+		objInput:           []string{`aa`},
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           [1]string{`aa`},
+		jsonExpectedOutput: `"aa"`,
+	},
+	{
+		objInput:           []string{`aa`, `bb`},
+		jsonExpectedOutput: `["set",["aa","bb"]]`,
+	},
+	{
+		objInput:           [2]string{`aa`, `bb`},
+		jsonExpectedOutput: `["set",["aa","bb"]]`,
+	},
+	{
+		objInput:           []int{10, 15},
+		jsonExpectedOutput: `["set",[10,15]]`,
+	},
+	{
+		objInput:           []uint16{10, 15},
+		jsonExpectedOutput: `["set",[10,15]]`,
+	},
```

Comment:
not sure why you are testing all simple types? if you test for string then no need to test for int, int32, float32, etc, etc. this adds no value as the focus here is to the OvsSet (which can contain anything) not serialization of all simple tests (something that is handled by standard json encoder/decoder).